framework/include/userobject/SolutionUserObject.h

```diff
@@ -75,6 +76,91 @@ class SolutionUserObject : public GeneralUserObject
   virtual Real pointValue(Real t, const Point & p, const unsigned int local_var_index) const;
 
   /**
+   * Returns a value at a specific location and variable (see SolutionFunction)
+   * @param t The time at which to extract (not used, it is handled automatically when reading the data)
+   * @param p The location at which to return a value
+   * @param var_name The variable to be evaluated
+   * @return The desired value for the given variable at a location
+   */
+  virtual Real pointValue(Real t, const Point & p, const std::string & var_name) const;
+
+  /**
+   * Returns a value at a specific location and variable for cases where the solution is
+   * multivalued at element faces
+   * Use pointValue for continuous shape functions or if you are sure your point is within an
+   * element!
+   * @param t The time at which to extract (not used, it is handled automatically when reading the data)
+   * @param p The location at which to return a value
+   * @param local_var_index The local index of the variable to be evaluated
+   * @return The desired value for the given variable at a location
+   */
+  virtual std::map<const Elem *, Real> discontinuousPointValue(Real t, Point pt, const unsigned int local_var_index) const;
```

Comment:
This map is actually fairly small, because it only stores one value for each element that is adjacent to one point. This could be 8 if it's a corner in a hex mesh e.g.